Assets/Scripts/PlayerScripts/PlayerSideAction.cs

```diff
@@ -0,0 +1,106 @@
+using Melanchall.DryWetMidi.Interaction;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class PlayerSideAction : PlayerAction
+{
+    public Melanchall.DryWetMidi.MusicTheory.NoteName noteRestriction2;
+    public List<double> timeStamps2 = new List<double>();
+    private int _inputIndex2;
+    private double _timeStamp2;
+
+    private double _curTimeStamp;
+    private int _curInputIndex;
+    public override void SetTimeStamps(IEnumerable<Note> array)
+    {
+        foreach (var note in array)
+        {
+            if (note.Octave == 1 && note.NoteName == noteRestriction)
+            {
+                var metricTimeSpan =
+                    TimeConverter.ConvertTo<MetricTimeSpan>(note.Time, MusicPlayer.MidiFileTest.GetTempoMap());
+                var spawnTime = ((double)metricTimeSpan.Minutes * 60f + metricTimeSpan.Seconds +
+                                 (double)metricTimeSpan.Milliseconds / 1000f);
+
+                timeStamps.Add(spawnTime - prespawnWarningSeconds);
+            }
+            else if (note.Octave == 1 && note.NoteName == noteRestriction2)
+            {
+                var metricTimeSpan =
+                    TimeConverter.ConvertTo<MetricTimeSpan>(note.Time, MusicPlayer.MidiFileTest.GetTempoMap());
+                var spawnTime = ((double)metricTimeSpan.Minutes * 60f + metricTimeSpan.Seconds +
+                                 (double)metricTimeSpan.Milliseconds / 1000f);
+
+                timeStamps2.Add(spawnTime - prespawnWarningSeconds);
+            }
+        }
+    }
+
+    // Update is called once per frame
+    protected override void Update()
```

Comment:
Why is this entire class overriding the main functionality of the parent class? I think this structure leads to a lot of duplicate code which isn't good. 
I think it would be better if one class handles player jumps and another class handles player side actions. You can also have both classes extend a parent class for any shared data.